### ISSUE TITLE
Special functions to Base

### ISSUE THREAD

**Issue by lindahua** [JuliaLang/julia#4301]
@andreasnoack implements a great collection of special functions for Distributions.jl. Some of these functions are probably useful outside of the distribution package. 

Source file: https://github.com/JuliaStats/Distributions.jl/blob/12d5aea1d9bea01e4883c120dc870dcf8912ecf9/src/specialfuns.jl

We have a discussion in issue https://github.com/JuliaStats/Distributions.jl/pull/148. Per @StefanKarpinski's suggestion, I open an issue here to invite broader attention.

Basically, we want to discuss: which of those functions should be moved to the Julia base.

**Comment by stevengj** [comment]
I feel like there should be a JuliaLang/SpecialFunctions.jl package that can be used as a repository for new special-function implementations.   Distributions.jl seems like the wrong place.

@nolta has an unregistered https://github.com/nolta/SpecialFunctions.jl package that might be a starting point.  (I feel like the https://github.com/nolta/Elliptic.jl pacakge should be folded into a larger special-functions package.)

I have a feeling that some of the more esoteric functions in Base, like the polygamma function, might be better off moved to such a package too.

I've also been playing with implementing exponential integrals (#7089) and [polylogarithms](https://en.wikipedia.org/wiki/Polylogarithm), which would be good for such a package.

**Comment by tkelman** [comment]
We should maybe create a new separate org for generic mathematically oriented packages. JuliaLang's CI resources are shared between base and all packages, so we should start migrating some of the packages here into organizations. Otherwise agree it's best not to tie these to Base's development cycle.

**Comment by johnmyleswhite** [comment]
The [StatsFuns](https://github.com/JuliaStats/StatsFuns.jl) package is a step in this direction, but I agree that we should just have an actual special functions package.

**Comment by stevengj** [comment]
@johnmyleswhite, StatsFun is fine for functions that are clearly limited to statistics, like the various CDF functions, but for more "generic" transcendental functions like `log1psq` it would be nice to move them to a separate package.

**Comment by simonbyrne** [comment]
+1 for a pure Julia SpecialFunctions.jl. Happy to help out, though probably don't have time to drive it.

**Comment by JeffBezanson** [comment]
Superseded by #8536 --- we should have a SpecialFunctions package.

**Comment by simonbyrne** [comment]
There is still the question of what should be moved out to a package (or stdlib, or whatever): the currently exported functions from `Base.Math` are:

* trigonmetric: 
  - libm: `sin`, `cos`, `tan`, `asin`, `acos`, `atan`,`atan2`
  - Julia: `sec`, `csc`, `cot`, `asec`, `acsc`, `acot`,
* hyperbolic: 
  - libm: `sinh`, `cosh`, `tanh`, `asinh`, `acosh`, `atanh`, 
  - Julia: `sech`, `csch`, `coth`, `asech`, `acsch`, `acoth`,
* sinc: 
  - Julia: `sinc`, `cosc`
* trig pi multiple:
  - Julia: `sinpi`, `cospi`,
* trig degree:
  - Julia: `cosd`, `cotd`, `cscd`, `secd`, `sind`, `tand`, `acosd`, `acotd`, `acscd`, `asecd`, `asind`, `atand`
* degree-radian conversions: 
  - Julia: `rad2deg`, `deg2rad`
* log: 
  - libm: `log`, `log2`, `log10`, `log1p`
* exp: 
  - libm: `exp`, `exp2`, `expm1`
  - Julia: `exp10`
* roots: 
  - intrinsic:`sqrt`
  - libm: `cbrt`
* error function: 
  - libm: `erf`, `erfc`
  - Faddeeva (openspecfun): `erfcx`, `erfi`, `dawson`
  - Julia: `erfinv`, `erfcinv`
* gamma: 
  - libm: `gamma`, `lgamma`
  - Julia: `lfact`, `beta`, `lbeta`, `digamma`, `trigamma`,`polygamma`, `invdigamma`, `eta`, `zeta`
* min/max: 
  - Julia: `max`, `min`, `minmax`, `clamp`, `clamp!`
* floating point specials: 
  - Julia: `exponent`, `significand`, `ldexp`, `frexp`,
* Airy functions: 
  - AMOS (openspecfun): `airy`, `airyai`, `airyprime`, `airyaiprime`, `airybi`, `airybiprime`, `airyx`
* Bessel functions: 
  - AMOS (openspecfun): various `besselX` and `hankelX` (there's a lot of these)
* misc:
  - libm: `modf`, `^`
  - Julia: `hypot`, `@evalpoly`
  - rem_pio2 (openspecfun): `mod2pi`

**Comment by JeffBezanson** [comment]
Categories I'd move to a package(s): error function, gamma, airy, bessel

**Comment by simonbyrne** [comment]
What about ones that also have mpfr (e.g. `bessely0` and `erf`)? Should we move those to the module?

**Comment by JeffBezanson** [comment]
Yes those can move to the package as well.

**Comment by ararslan** [comment]
In reference to what should be moved, see also https://github.com/JuliaMath/SpecialFunctions.jl/issues/12.

**Comment by musm** [comment]
In https://github.com/JuliaLang/julia/pull/19560  I have moved all the special functions to their own module `SpecFun` and have left the elementary function in the `Math` module
